pkg/operator/configobserver/oauth/observe_tokenconfig.go

```diff
@@ -0,0 +1,129 @@
+package oauth
+
+import (
+	"fmt"
+
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/klog"
+
+	configlistersv1 "github.com/openshift/client-go/config/listers/config/v1"
+	"github.com/openshift/library-go/pkg/operator/configobserver"
+	"github.com/openshift/library-go/pkg/operator/events"
+)
+
+// OAuthLister lists OAuth information
+type OAuthLister interface {
+	OAuthLister() configlistersv1.OAuthLister
+}
+
+const (
+	defaultAccessTokenMaxAgeSeconds   = float64(86400) // a day
+	fieldAccessTokenMaxAgeSeconds     = "accessTokenMaxAgeSeconds"
+	fieldAccessTokenInactivityTimeout = "accessTokenInactivityTimeout"
+)
+
+// ObserveAccessTokenMaxAgeSeconds returns an unstructured fragment of KubeAPIServerConfig that changes the default value for access token max age,
+// if there is a valid value for it in OAuth cluster config.
+func ObserveAccessTokenMaxAgeSeconds(genericlisters configobserver.Listers, recorder events.Recorder, existingConfig map[string]interface{}) (ret map[string]interface{}, errs []error) {
```

Comment:
no named parameters.  It makes the function harder to reason about because I have to track current state.

Reply:
See below about pruning. We fixed all library-go observers months ago and this style was the only one we didn't scream about. Every other error logic reaching the same goal was error prone and ugly.